app/controllers/concerns/json_pagination.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module JsonPagination
+  private
+
+  def add_pagination_links(json, pagy)
+    # add custom headers to the response
+    add_headers(pagy)
+
+    # Insert links at beginning
+    links = links(pagy)
+    if links.present?
+      json_d = json.clone
+      json.clear
+      json["links"] = links
+      json.merge! json_d
```

Comment:
just add them add the end ?

Reply:
Generally, we can add anywhere,  but some of our API results are very large along with `includes` and `inlines` its very hard to find it out while viewing in UI and also it's hard to maintain the `links` always as the last entry.. something may end-up up append in between.

Reply:
then how about:
```ruby
json = {links: links}.merge(json)
```

also document this reason why we are doing it